### PR TITLE
[addons] change the use of CBinaryAddonBase

### PR DESCRIPTION
**Pull request by AlwinEsch**
## Description
This removes the parts that have not been needed since cpluff removal on the Binary Addon Interface.

At the moment the loading of the addon.xml is still done separately with binary addons, since this could not be avoided beforehand with cpluff.

## How Has This Been Tested?
Here at home with some addons, but must be specially tested on the input stream addons synonymous of others!

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/addons/binary-addons/BinaryAddonBase.h [hunk 1]
```diff
@@ -32,17 +29,15 @@ namespace ADDON
...
     const std::string& ID() const { return m_addonInfo->ID(); }
     const std::string& Path() const { return m_addonInfo->Path(); }
 
     TYPE MainType() const { return m_addonInfo->MainType(); }
     const std::string& MainLibName() const { return m_addonInfo->LibName(); }
 
-    bool IsType(TYPE type) const;
-    const std::vector<CBinaryAddonType>& Types() const { return m_types; }
-    const CBinaryAddonType* Type(TYPE type) const;
+    bool IsType(TYPE type) const { return m_addonInfo->IsType(type); }
```
Comment (garbear) [inline xbmc/addons/binary-addons/BinaryAddonBase.h]: I like to keep lines that require knowledge of implementation details in the .cpp file. Even if it's just one line, this prevents the type from being forward-declared in the future.

**Review by notspiff — APPROVED** [review]
Lots of lovely red.

File: xbmc/addons/binary-addons/BinaryAddonBase.h (hunk 1 above)
Comment (AlwinEsch) [reply xbmc/addons/binary-addons/BinaryAddonBase.h]: Have moved them to the cpp

File: xbmc/addons/binary-addons/BinaryAddonBase.h [hunk 2]
```diff
@@ -32,44 +29,36 @@ namespace ADDON
...
-
-    const std::string& ID() const { return m_addonInfo->ID(); }
-    const std::string& Path() const { return m_addonInfo->Path(); }
+    const std::string& ID() const;
+    const std::string& Path() const;
 
-    TYPE MainType() const { return m_addonInfo->MainType(); }
-    const std::string& MainLibName() const { return m_addonInfo->LibName(); }
+    TYPE MainType() const;
+    const std::string& MainLibName();
```
Comment (Rechi) [inline xbmc/addons/binary-addons/BinaryAddonBase.h]:
```suggestion
    const std::string& MainLibName() const;
```

File: xbmc/addons/binary-addons/BinaryAddonBase.cpp [hunk 3]
```diff
@@ -17,58 +17,92 @@
...
+const ArtMap& CBinaryAddonBase::Art() const
 {
-  int contents = 0;
-  for (auto addonType : m_types)
-    contents += addonType.ProvidedSubContents();
-  return (contents > 0);
+  return m_addonInfo->Art();
+}
+
+const std::string& CBinaryAddonBase::Disclaimer() const { return m_addonInfo->Disclaimer();
```
Comment (Rechi) [inline xbmc/addons/binary-addons/BinaryAddonBase.cpp]:
```suggestion
const std::string& CBinaryAddonBase::Disclaimer() const
{
  return m_addonInfo->Disclaimer();
```

**Pull request merged**